src/content/articles/why-typescript-doesnt-follow-strict-semantic-versioning.mdx

```diff
@@ -0,0 +1,300 @@
+- A new _major_ version of TypeScript (e.g. 5.0.0) is released if the minor version would have exceeded 9 (e.g. 4.10.0)
+
+In other words, TypeScript uses _major_ version numbers more for marketing and public visibility than for semantic versioning.
+This was an intentional engineering decision: releasing a new version of TypeScript almost always requires compiler changes that "break" some existing TypeScript programs.
+
+### "Breaking" Type Checking Changes
+
+Many users consider TypeScript's type checking as a part of its API.
+When a minor or patch version introduces changes to type checkin, some might consider that a "breaking change".
+
+But - the type checker is a core part of TypeScript, and one of the most frequently modified parts of each release.
+Strictly following semver would mean any change to type checking would be considered "breaking".
+
+Changes to the type system can come in many forms.
+Even seemingly small changes in each of those forms can cause breakages for some consumers of TypeScript.
+
+<CaptionedImage
+  alt="xkcd 'Workflow' comic from the linked page"
+  id="xkcd-workflow"
+  label="This famous xkcd comic is humorous, but for many projects is not far off from reality."
+  href="https://xkcd.com/1172"
+  src="https://imgs.xkcd.com/comics/workflow_2x.png"
+/>
+
+#### Added Type Errors
+
+One of the main benefits of using TypeScript is that it reports a type error when it detects an issue in code.
+New versions of TypeScript are often made more capable in finding more classses of type errors.
+But what some would consider bugfixes or new features in type checking, others would consider introduced breakages.
+
+Updating to a new version of TypeScript and seeing new red editor squigglies or build failures could be considered "breaking" - but is a natural consequence of TypeScript improving over time.
+
+Example: [TypeScript 4.8 stopped allowing unconstrained generic type parameters to be assignable to `{}`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-8.html#unconstrained-generics-no-longer-assignable-to-).
+This is technically correct, as `{}` is not supposed to be assignable to `null` or `undefined`.
+But existing code might have been written with the assumption that the type parameters would always be provided with an object type.
+That code would newly receive type errors in TypeScript >=4.8.
+
+#### Removed Type Errors
+
+Similarly, TypeScript sometimes stops reporting type errors in situations that used to be considered incorrect.
+This is sometimes done to make TypeScript more permissive and compatible with existing JavaScript code.
+But for consumers who relied on TypeScript's stricter type checking to catch issues, these changes could also be considered "breaking".
+
+Updating to a new version of TypeScript and no longer being prevented from shipping code patterns the project wanted to avoid could also be considered "breaking" - even if TypeScript considers the change an improvement.
```

Comment:
[nitpick] What is this paragraph describing here? It could use some clarity. 
```suggestion
Updating to a new version of TypeScript and no longer being prevented from shipping code patterns the project wanted to avoid could also be considered "breaking" - even if TypeScript considers the change an improvement.
```